commands.js

```diff
@@ -1485,6 +1489,7 @@ exports.commands = {
 			let modchat = Tools.escapeHTML(room.modchat);
 			this.add("|raw|<div class=\"broadcast-red\"><b>Moderated chat was set to " + modchat + "!</b><br />Only users of rank " + modchat + " and higher can talk.</div>");
 		}
+		if (room.battle && !room.modchat && !user.can('modchat')) room.modchatUser = '';
```

Comment:
Having `room.requestModchat()` modify `room.modchatUser` but then editing it directly here is confusing